src/main/java/Nehsik.java

```diff
@@ -0,0 +1,127 @@
+                unmarkTask(command, taskList);
+            }  else if (command.startsWith("todo")) {
+                addTodoTask(command, taskList, currTaskCount);
+                currTaskCount = acknowledgeTaskAdded(taskList, currTaskCount);
+            } else if (command.startsWith("deadline")) {
+                addDeadlineTask(command, taskList, currTaskCount);
+                currTaskCount = acknowledgeTaskAdded(taskList, currTaskCount);
+            } else if (command.startsWith("event")) {
+                addEventTask(command, taskList, currTaskCount);
+                currTaskCount = acknowledgeTaskAdded(taskList, currTaskCount);
+            } else if (command.equals("bye")) {
+                displayExitMessage();
+                break;
+            } else {
+                System.out.println("Invalid Command");
+            }
+        }
+
+        in.close();
+    }
+
+    private static void displayTaskList(int currTaskCount, Task[] taskList) {
+        printLine();
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < currTaskCount; i++) {
+            System.out.println((i + 1) + "." + taskList[i].toString());
+        }
+        printLine();
+    }
+
+    private static void markTask(String command, Task[] taskList) {
+        printLine();
+        System.out.println("Nice! I've marked this task as done:");
+        int taskNum = Integer.parseInt(command.substring(MARK_TASK_INDEX)) - 1;
+        taskList[taskNum].markAsDone();
+        System.out.println(taskList[taskNum].toString());
+        printLine();
+    }
+
+    private static void unmarkTask(String command, Task[] taskList) {
+        printLine();
+        System.out.println("OK, I've marked this task as not done yet:");
+        int taskNum = Integer.parseInt(command.substring(UNMARK_TASK_INDEX)) - 1;
+        taskList[taskNum].markAsUndone();
+        System.out.println(taskList[taskNum].toString());
+        printLine();
+    }
+
+    private static void addTodoTask(String command, Task[] taskList, int currTaskCount) {
+        String taskDescription = command.substring(TODO_DESCRIPTION_POSITION).trim();
+        taskList[currTaskCount] = new Todo(taskDescription);
+    }
+
+    private static void addDeadlineTask(String command, Task[] taskList, int currTaskCount) {
+        int descriptionStartPosition = command.indexOf("deadline ") + 9;
+        int descriptionEndPosition = command.indexOf("/by ") - 1;
+        String taskDescription = command.substring(descriptionStartPosition, descriptionEndPosition).trim();
+
+        int byStringPosition = command.indexOf("/by ") + 4;
+        String by = command.substring(byStringPosition).trim();
+
+        taskList[currTaskCount] = new Deadline(taskDescription, by);
+    }
+
+    private static void addEventTask(String command, Task[] taskList, int currTaskCount) {
+        int descriptionStartPosition = command.indexOf("event ") + 6;
```

Comment:
Perhaps you would want to use constants rather than [magic numbers](Kishen271828).